anomalib/models/cfa/torch_model.py

```diff
@@ -0,0 +1,387 @@
+"""Torch Implementatation of the CFA Model.
+
+CFA: Coupled-hypersphere-based Feature Adaptation for Target-Oriented Anomaly Localization
+
+Paper https://arxiv.org/abs/2206.04325
+"""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, List, Optional, Tuple, Union
+
+import torch
+import torch.nn.functional as F
+import torchvision
+from einops import rearrange
+from sklearn.cluster import KMeans
+from torch import Tensor, nn
+from torch.fx.graph_module import GraphModule
+from torch.nn.common_types import _size_2_t
+from torch.utils.data import DataLoader
+from torchvision.models.feature_extraction import create_feature_extractor
+from tqdm import tqdm
+
+from anomalib.models.components import DynamicBufferModule, GaussianBlur2d
+
+SUPPORTED_BACKBONES = ("vgg19_bn", "resnet18", "wide_resnet50_2", "efficientnet_b5")
+
+
+# TODO: Replace this with the new torchfx feature extractor.
```

Comment:
Maybe we can add https://github.com/openvinotoolkit/anomalib/blob/main/anomalib/models/components/feature_extractors/torchfx.py here and switch to feature extractor when the feature branch is merged.